File: solutions/Javascript/Remove Linked List Elements.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

var removeElements = function (head, val) {
  let prev = null;
  let current = head;
  while (current) {
    if (head.val === val) {
      head = current.next;
      prev = head;
      current = current.next;
    } else if (current.val === val) {
      prev.next = current.next;
      current = current.next;
    } else {
      prev = current;
      current = current.next;
    }
  }

  return head;
};

File: solutions/cpp/Remove Linked List Elements.cpp
#include <iostream>
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    ListNode *removeElements(ListNode *head, int val)
    {
        ListNode *prev = NULL, *current = head;
        while (current)
        {
            if (head->val == val)
            {
                head = head->next;
                prev = head;
                current = current->next;
            }
            else if (current->val == val)
            {
                prev->next = current->next;
                current = current->next;
            }
            else
            {
                prev = current;
                current = current->next;
            }
        }
        return head;
    }
};